ctms/acoustic_service.py

```diff
@@ -0,0 +1,279 @@
+        "common-voice": "sub_common_voice",
+        "mozilla-fellowship-awardee-alumni": "sub_mozilla_fellowship_awardee_alumni",
+        "firefox-accounts-journey": "sub_firefox_accounts_journey",
+        "firefox-news": "sub_firefox_news",
+        "mozilla-and-you": "sub_firefox_news",
+        "hubs": "sub_hubs",
+        "internet-health-report": "sub_internet_health_report",
+        "knowledge-is-power": "sub_knowledge_is_power",
+        "miti": "sub_miti",
+        "mixed-reality": "sub_mixed_reality",
+        "mozilla-festival": "sub_mozilla_festival",
+        "mozilla-foundation": "sub_mozilla_foundation",
+        "mozilla-technology": "sub_mozilla_technology",
+        "mozillians-nda": "sub_mozillians_nda",
+        "take-action-for-the-internet": "sub_take_action_for_the_internet",
+        "test-pilot": "sub_test_pilot",
+        "about-mozilla": "sub_about_mozilla",
+        "app-dev": "sub_apps_and_hacks",
+        "firefox-welcome": "firefox-welcome",  # TODO: Confirm approach, found newsletters through testing
+        "mozilla-welcome": "mozilla-welcome",  # TODO: Confirm
+        "firefox-os": "firefox-os",  # TODO: Confirm
+        "ambassadors": "ambassadors",  # TODO: Confirm
+        "maker-party": "maker-party",  # TODO: Confirm
+        "mozilla-learning-network": "mozilla-learning-network",  # TODO: Confirm
+    }
+
+
+class Acoustic(Silverpop):
+    """
+    :param client_id
+    :param client_secret
+    :param refresh_token
+    :param server_number
+    """
+
+    @staticmethod
+    def _process_response(resp):
+        response = etree.fromstring(resp.text.encode("utf-8"))
+        # success = response.find(".//SUCCESS")
+
+        # how main table failures are reported:
+        fault = response.find(".//Fault/FaultString")
+        if fault is not None:
+            # print(fault.text) # TODO: Convert to logging
+            raise SilverpopResponseException(fault.text)
+
+        # how RT failures are reported:    (hooray for consistency.)
+        failures = response.findall(".//FAILURES/FAILURE")
+        if failures is not None:
+            for fail in failures:
+                raise SilverpopResponseException(etree.tostring(fail))
+
+        return response
+
+    def _call(self, xml):
+        # Following basket's suggested pattern,
+        # adding force_bytes to deal with some Acoustic rejections
+        response = self.session.post(self.api_endpoint, data={"xml": force_bytes(xml)})
+        return self._process_response(response)
+
+
+class CTMSToAcousticService:
+    def __init__(
+        self,
+        client_id,
+        client_secret,
+        refresh_token,
+        acoustic_main_table_id,
+        acoustic_newsletter_table_id,
+        server_number=6,
+    ):
+        """
+        Construct a CTMSToAcousticService object that can interface between contacts and acoustic forms of data
+        """
+        self.acoustic = Acoustic(
```

Comment:
Nit:I think a cleaner interface would be to pass an ``Acoustic`` instance to ``CTMSToAcousticService``, rather than to construct it in the ``__init__`` method. For example, it would allow you to pass a mock object for tests.

Reply:
I believe in the recent commit: c42ccef86b1835e4a6b337946166ca09827ccde7

I made the changes requested; however, I feel that I need to still tweak the behavior of the way I'm using mocks in tests.